### PR TITLE
Add a check for optimizer attached to lr_scheduler

### PR DESCRIPTION
**Pull request by rohitgr7**
## What does this PR do?

Added an exception in case the optimizer attached to a scheduler is not returned in `configure_optimizers`.

## Before submitting
- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes [if needed]?
- [x] Did you write any new necessary tests [no need for typos, docs]? 
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:  

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified
 - [x] **Check that target branch and milestone are aligned!**
 

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/trainer/optimizers.py [hunk 1]
```diff
@@ -183,3 +185,10 @@ def zero_grad(self):
 
     def __repr__(self):
         return 'No Optimizer'
+
+
+def _validate_scheduler_optimizer(optimizers, lr_schedulers):
+    if any(sch['scheduler'].optimizer not in optimizers for sch in lr_schedulers):
+        raise MisconfigurationException(
+            "Some schedulers are attatched with an optimizer that wasn't returned from `configure_optimizers`."
```
Comment (awaelchli) [inline pytorch_lightning/trainer/optimizers.py:193]:
```suggestion
            "Some schedulers are attached with an optimizer that wasn't returned from `configure_optimizers`."
```

File: tests/trainer/test_optimizers.py [hunk 2]
```diff
@@ -483,3 +483,20 @@ def test_lr_scheduler_with_no_actual_scheduler_raises(tmpdir):
...
+    class InvalidOptimizerModel(BoringModel):
+        def configure_optimizers(self):
+            opt1 = torch.optim.SGD(self.layer.parameters(), lr=0.1)
+            opt2 = torch.optim.SGD(self.layer.parameters(), lr=0.1)
+            lr_scheduler = torch.optim.lr_scheduler.StepLR(opt2, step_size=1)
+            return [opt1], [lr_scheduler]
+
+    model = InvalidOptimizerModel()
+    trainer = Trainer(default_root_dir=tmpdir, fast_dev_run=True)
+    with pytest.raises(MisconfigurationException, match="attatched with an optimizer that wasn't returned"):
```
Comment (awaelchli) [inline tests/trainer/test_optimizers.py:501]:
```suggestion
    with pytest.raises(MisconfigurationException, match="attached with an optimizer that wasn't returned"):
```

File: tests/trainer/test_optimizers.py [hunk 3]
```diff
@@ -483,3 +483,20 @@ def test_lr_scheduler_with_no_actual_scheduler_raises(tmpdir):
     trainer = Trainer(default_root_dir=tmpdir, fast_dev_run=True)
     with pytest.raises(MisconfigurationException, match='The lr scheduler dict must have the key "scheduler"'):
         trainer.fit(model)
+
+
+def test_invalid_optimizer_in_scheduler(tmpdir):
+    """
+    Test exception when optimizer attatched to lr_schedulers wasn't returned
```
Comment (awaelchli) [inline tests/trainer/test_optimizers.py:490]:
```suggestion
    Test exception when optimizer attached to lr_schedulers wasn't returned
```

File: pytorch_lightning/trainer/optimizers.py (hunk 1 above)
Comment (awaelchli) [reply pytorch_lightning/trainer/optimizers.py:193]: should we avoid contractions in messages and docs?

**Review by awaelchli — APPROVED** [review]

File: pytorch_lightning/trainer/optimizers.py (hunk 1 above)
Comment (rohitgr7) [reply pytorch_lightning/trainer/optimizers.py]: yeah maybe. it's not recommended?

Comment (awaelchli) [reply pytorch_lightning/trainer/optimizers.py]: not sure, maybe just personal preference but in scientific works contractions are usually avoided. 
maybe we want to deal with this another time:)

**Comment by carmocca** [comment]
Are we sure this won't break some very-specific scheduler use case where an external optimizer is used?

**Review by s-rog — APPROVED** [review]

**Comment by rohitgr7** [comment]
> Are we sure this won't break some very-specific scheduler use case where an external optimizer is used?

by external optimizer what do you mean? @carmocca

**Comment by carmocca** [comment]
I mean an optimizer that is not managed by Lightning through `configure_optimizers`. I'm not sure if anybody is doing it but this PR would break their workflow.

The question is, should we make this a warning instead or is it safe to merge?

**Review by tchaton — APPROVED** [review]
Great catch !

**Comment by rohitgr7** [comment]
> I mean an optimizer that is not managed by Lightning through configure_optimizers. I'm not sure if anybody is doing it but this PR would break their workflow.

not sure if anyone would do that. but let's see if we see someone raise an issue for this in the future. We can definitely change it to a warning later.

**Review by Borda — APPROVED** [review]
add chlog pls

**Review by carmocca — APPROVED** [review]
